docs/pages/machine-id/deployment/spacelift.mdx

```diff
@@ -0,0 +1,286 @@
+---
+title: Deploying Machine ID on Spacelift
+description: How to install and configure Machine ID on Spacelift
+---
+
+You can use Spacelift with the Teleport Terraform provider to manage dynamic
+configuration resources via GitOps and infrastructure as code. This gives you an
+audit trail of changes to your Teleport configuration and a single source of
+truth for operators to examine.
+
+This guide shows you how to use the Machine ID agent, `tbot`, to allow the
+Teleport Terraform provider running in a Spacelift stack to configure your
+Teleport cluster.
+
+In this setup, `tbot` proves its identity to the Teleport Auth Service by
+presenting an ID token signed by Spacelift. This allows `tbot` to authenticate
+with the Teleport cluster without the need for a long-lived shared secret.
+
+While following this guide, you will create a Teleport user and role with no
+privileges in order to show you how to use Spacelift to create dynamic
+resources. 
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- (!docs/pages/includes/tctl.mdx!)
+- A GitHub repository where you will store your Terraform configuration and a
+  Spacelift stack linked to this repository.
+- A paid Spacelift account. This is required to use the `spacelift` join method.
```

Comment:
Should we make Spacelift joining enterprise only?

Similar to what we did with GitHub Enterprise - if a Teleport integration requires paid 3rd-party software, the Teleport feature itself should probably be for paid customers too.

WDYT?